### PR TITLE
feat(FluidDatePicker): implement `FluidDatePicker`

### LINKED ISSUE #12403: Fluid inputs: Add FluidDatePicker experimental implementation
- [x] Place implementation within a new component, `FluidComponentName`
  - [x] Add VRT tests via storybook story e2e setup
  - [x] Add unit tests (likely simple, eg. "does the classname get added")
  - [x] Add storybook example
  - [x] Add it to the FluidForm story
  - [x] Ensure styles are properly scoped to their own file
  - [ ] Add `component: FluidDatePicker ` label to repo for triaging issues

### PR DESCRIPTION
**Pull request by tw15egan**
Refs https://github.com/carbon-design-system/carbon/issues/12125
Closes https://github.com/carbon-design-system/carbon/issues/12403

Adds in `unstable__FluidDatePicker` and `unstable__FluidDatePickerInput`

#### Changelog

**New**

- `unstable__FluidDatePicker` component
- Styles for `FluidDatePicker` (`_fluid-date-picker.scss`)
- Stories for `FluidDatePicker` (`FluidDatePicker.stories.js`)
- Tests for `FluidDatePicker`

**Changed**

- Fixed some styles I noticed when adding in `FluidDatePicker` to the `FluidForm` story

#### Testing / Reviewing

Go to `FluidDatePicker` and check out the stories. For now, I've got all the different states in (they will be removed before merging). Ensure all playground controls are correct and stories render as expected.

### CODE REVIEW

**Comment by aagonzales** [comment]
@tw15egan yeah we should probably allow just triggering invalid on one side ... I'm not yet sure how that will look visually. We might have to come back to that.

Also kind of on the same thread, I forget to spec out `timePicker` in fluid and will have a sam joint input problem.

**Comment by tw15egan** [comment]
@aagonzales, so would the invalid outline span the entire input, but just keep invalid text to one input (as-is) for now? Since it's experimental we can come back and refine this interaction as well

**Comment by aagonzales** [comment]
@tw15egan I'm going to talk about this in design crit tomorrow and hopefully get a final answer. The bottom+left is how in forms, we originally said we would treat inline fluid inputs with one invalid state. 

image

**Comment by aagonzales** [comment]
@tw15egan This is what we decided, is it doable?

image

**Comment by tw15egan** [comment]
Yeah, we should be able to do that; we'll just need to add a prop of `invalid` to the `DatePicker` in addition to the `DatePickerInput`. Would this also be a valid use case outside of Fluid components? i.e., the entire `DatePicker` is invalid? If so, we may want to add that to the normal `DatePicker` as well

**Comment by tw15egan** [comment]
I believe AVT tests are failing due to the duplicate ids on the stories showcasing all the different warning/error variants. Before merging, these will all be removed and set to a single story

**Review by aagonzales — COMMENTED** [review]
The componet styles all look correct but there are some bugs in the Playground.
- I'm getting some weird bugs when I switch between Range and Single.
   - Switch from single back to range and I can't get the end date calendar menu to open
   - Switch from range (with a range selected) to single, the range is still showing when I open the menu.
- When invalid is turned on the width of the inputs grow. The error message should be wrapping and not effecting the width. (Same for warning)
Oct-28-2022 13-49-45

**Comment by tw15egan** [comment]
That switcher is a bit temperamental, so I switched it to show two components. Also, I set a width on the container, so the component expands downwards now.

File: packages/react/src/components/FluidDatePicker/FluidDatePicker.stories.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      disable: true,
+    },
+  },
+  locale: {
+    table: {
+      disable: true,
+    },
+  },
+  onChange: {
+    action: 'clicked',
```
Comment (tay1orjones) [inline packages/react/src/components/FluidDatePicker/FluidDatePicker.stories.js:167]:
```suggestion
    action: 'clicked',
    table: {
      disable: true,
    },
```

File: packages/react/src/components/FluidDatePicker/FluidDatePicker.stories.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  },
+  onClose: {
+    action: 'clicked',
```
Comment (tay1orjones) [inline packages/react/src/components/FluidDatePicker/FluidDatePicker.stories.js:173]:
```suggestion
    action: 'clicked',
    table: {
      disable: true,
    },
```

File: packages/react/src/components/FluidDatePicker/FluidDatePicker.stories.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  },
+  onOpen: {
+    action: 'clicked',
```
Comment (tay1orjones) [inline packages/react/src/components/FluidDatePicker/FluidDatePicker.stories.js:179]:
```suggestion
    action: 'clicked',
    table: {
      disable: true,
    },
```

**Review by tay1orjones — APPROVED** [review]
Looks great to me! 🎉 Just some minor suggestions for the playground story

**Review by aagonzales — APPROVED** [review]
Looks good to go! 👍 ✅

**Review by alisonjoseph — APPROVED** [review]

**Pull request merged**